pkg/kubelet/kubelet.go

```diff
@@ -1052,7 +1052,10 @@ func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *api.ObjectFieldSelector, pod
 	if err != nil {
 		return "", err
 	}
-
+	switch internalFieldPath {
+	case "status.podIP":
```

Comment:
Is the reason this can't go into ExtractFieldPathAsString because that function is the same for all resource types? A comment here to that effect would be useful.